src/storage/src/render/debezium.rs

```diff
@@ -159,13 +157,50 @@ where
         } => panic!("render_tx should be called with a dedup mode"),
     };
 
-    // TODO(#11666): Should we try to avoid moving everything to one worker?
+    let DebeziumTransactionMetadata {
+        tx_data_collections_idx,
+        tx_data_collections_data_collection_idx,
+        tx_data_collection_name,
+        tx_data_collections_event_count_idx,
+        tx_status_idx,
+        tx_transaction_id_idx,
+        data_transaction_id_idx,
+        tx_metadata_global_id: _,
+    } = tx_metadata_description;
+
+    let tx_dist = move |&(ref row, time, _diff): &(Row, Timestamp, Diff)| match row
+        .iter()
+        .nth(tx_transaction_id_idx)
+    {
+        Some(Datum::String(s)) => s.hashed(),
+        // If there's no transaction_id then we won't be able to use this row so aim to distribute evenly
+        _ => time.hashed(),
+    };
+
+    let data_dist = move |result: &DecodeResult| {
+        // If we can't pull out the transaction_id, it doesn't matter which worker we end up on.  Position seems a good
+        // way to spread around the work as it comes in and is likely uncorrelated to the hash.
+        let default_hash = result.position.hashed();
+        let value = match result.value.as_ref() {
+            Some(Ok(v)) => v,
+            _ => return default_hash,
+        };
+        let transaction = match value.iter().nth(transaction_idx) {
+            Some(Datum::List(l)) => l,
+            _ => return default_hash,
+        };
+        match transaction.iter().nth(data_transaction_id_idx) {
```

Comment:
dont we `unwrap` these later on? the _ case here, and panic on anything but `Null` or `List`, should we match that here?

Reply:
yeah makes sense -- will be more future-proof as any changes to what shapes we allow will require us to fix up here and thus properly distribute messages.

...as much as I don't like the unwraps / panics of types littered around

Reply:
I also value consistency over no unwraps, but definitely possible its worth a cleanup issue here!